<?xml version="1.0"?>
<launch>

    <!-- File names of setting files. -->
    <arg name="supereight_ros_config_file"
         default="/home/anna/catkin_exploration_ws/src/exploration_ros/planner/supereight_ros/data/config.yaml"/>
    <arg name="depth_image_topic"
         default="/firefly/vi_sensor/camera_depth/depth/disparity"/>
    <arg name="pose_topic" default="/firefly/vi_sensor/ground_truth/transform"/>
    <arg name="camera_info_topic" default="/firefly/vi_sensor/camera_depth/depth/camera_info"/>
    <arg name="enable_icp_tracking_" default="false"/>

    <arg name="use_tf_transforms_" default="true"/>
    <arg name="body_frame" default="firefly/base_link"/>
    <arg name="tf_frame" default="navigation"/>
    <!--<arg name="depth_image_topic"-->
    <!--default="/camera/aligned_depth_to_color/image_raw"/>-->

    <!--<arg name="pointcloud_topic" default="/firefly/vi_sensor/camera_depth/depth/points"/>-->
    <!--<arg name="pose_topic" default="/vicon/d435/d435"/>-->
    <!--<arg name="camera_info_topic" default="/firefly/vi_sensor/camera_depth/depth/camera_info"/>-->
    <!-- Launch ptp_ros node -->
    <node pkg="supereight_ros" name="supereight"
          type="supereight_ros" clear_params="true" output="screen"
          launch-prefix="gdb -ex run -args">
        <!-- valgrind -v -track-origins=yes -leak-check=fulaunch-prefix="gdb -ex run -args"->
        <!- Include rosparam yaml files -->
        <rosparam command="load" file="$(arg supereight_ros_config_file)"/>

        <remap from="/camera/camera_info" to="$(arg camera_info_topic)"/>

        <!-- for other launch files remap camera image topic to this image topic-->
        <remap from="/camera/depth_image" to="$(arg depth_image_topic)"/>
        <!-- for other launch files remap pose estimation topic to this pose
        topic-->
        <remap from="/pose" to="$(arg pose_topic)"/>
        <param name="tf_frame" type="string" value="$(arg tf_frame)"/>
        <param name="body_frame" type="string" value="$(arg body_frame)"/>
        <param name="enable_icp_tracking_" type="bool" value="$(arg enable_icp_tracking_)"/>
        <param name="use_tf_transforms_" type="bool" value="$(arg use_tf_transforms_)"/>
    </node>
</launch>
